src/credentials/BUILD.gn

```diff
@@ -36,11 +36,11 @@ static_library("credentials") {
     "FabricTable.h",
     "GenerateChipX509Cert.cpp",
     "GroupDataProvider.h",
+    "GroupDataProviderImpl.cpp",
```

Comment:
This ties the dependency here. We should make the example implementations live in different build targets otherwise there is no way to override and we lose the benefit of the interface

Reply:
Agreed. [This ticket should take care of it](https://github.com/project-chip/connectedhomeip/issues/12276).